edb/server/compiler/compiler.py

```diff
@@ -927,6 +959,31 @@ def _compile_and_apply_ddl_stmt(
                 single_unit=False,
             )
 
+        # If we are in a migration rewrite, we also don't actually
+        # apply the DDL, just record it. (The DDL also needs to be a
+        # CreateMigration.)
+        if mrstate := current_tx.get_migration_rewrite_state():
+            if not isinstance(stmt, qlast.CreateMigration):
```

Comment:
Not necessary in this iteration, but in principle we can allow any DDL statements and automatically collect them into a migration.

Reply:
What typically happens in practice is that a lone DDL statement will get automatically put into its own migration